### PR TITLE
wpcom-block-editor: Track the command palette behavior

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/issues/94267

## Proposed Changes

* Add the following events to the command palette

**wpcom_editor_command_palette_open**
image

**wpcom_editor_command_palette_search**
image

**wpcom_editor_command_palette_selected**
image

**wpcom_editor_command_palette_close**
image

## Why are these changes being made?

* We want to understand the discoverability of the Command Palette

## Testing Instructions

* Go to the Editor
* On the sidebar of the Site Editor
  * Click the "Search" icon and make sure the `wpcom_editor_command_palette_open` event is sent
  * Type something and make sure the `wpcom_editor_command_palette_search` event is sent with the `keyword` property
  * Select any result and make sure the `wpcom_editor_command_palette_selected` event is sent with the `value` property
  * Make sure the `wpcom_editor_command_palette_close` event is sent when the command palette is closed
* On the edit mode of the Site Editor or page/post editor
  * Click the title section with ( cmd + K ) in the header to open the command palette
      image
  * Make sure the `wpcom_editor_command_palette_open` event is sent
  * Repeat above steps and make sure the following events are sent correctly
    * `wpcom_editor_command_palette_search`
    * `wpcom_editor_command_palette_selected`
    * `wpcom_editor_command_palette_close`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js [hunk 1]
```diff
@@ -798,6 +806,72 @@ const trackInstallBlockType = ( block ) => {
...
+		tracksRecordEvent( 'wpcom_editor_command_palette_selected', {
+			value: selectedCommandElement.dataset.value,
+		} );
+	}
+};
+
+const trackCommandPaletteOpen = () => {
+	tracksRecordEvent( 'wpcom_editor_command_palette_open' );
+
+	window.setTimeout( () => {
```
Comment (xavier-lc) [inline apps/wpcom-block-editor/src/wpcom/features/tracking.js:838]: Why the need for `setTimeout`?

**Review by xavier-lc — APPROVED** [review]
Works as described 👌

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js (hunk 1 above)
Comment (arthur791004) [reply apps/wpcom-block-editor/src/wpcom/features/tracking.js:838]: I want to make sure the modal is there before binding the events

Comment (xavier-lc) [reply apps/wpcom-block-editor/src/wpcom/features/tracking.js:838]: Makes sense. Maybe a `MutationObserver` would be safer? No need to spend time on that if it's currently working, though.

**Pull request merged**